packages/dds/map/src/directory.ts

```diff
@@ -1074,22 +1119,26 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
         }
 
         // Create the sub directory locally first.
-        const isNew = this.createSubDirectoryCore(subdirName, true);
-
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const subDir: IDirectory = this._subdirectories.get(subdirName)!;
+        const isNew = this.createSubDirectoryCore(subdirName, true, -1, this.directory.isAttached() ?
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            this.runtime.clientId! : "detached");
+        const subDir = this._subdirectories.get(subdirName);
+        assert(subDir !== undefined, "subdirectory should exist after creation");
 
         // If we are not attached, don't submit the op.
         if (!this.directory.isAttached()) {
             return subDir;
         }
 
-        const op: IDirectoryCreateSubDirectoryOperation = {
-            path: this.absolutePath,
-            subdirName,
-            type: "createSubDirectory",
-        };
-        this.submitCreateSubDirectoryMessage(op, !isNew);
+        // Only submit the op, if it is newly created.
+        if (isNew) {
```

Comment:
I see two other possible designs:
- Create fails if sub-directory exists
- Create overwrites previous content (i.e. same as delete + create).
I think current design of silently doing nothing is wrong. Most likely, users will assume that after creating they deal with empty directory, which is not the case. Maybe other way to say it - if users run into such condition (of creating existing directory), they probably already need to have some special handling for such case, and correctly written code should not run into such cases. Failing (or overwriting) seems like helps people find opposite cases and forces users to think about proper handling

Reply:
Creating an issue for this and will follow up in next branch as this would be a breaking change. I think I like the first approach more, to just fail the creation and the user can do whatever they want instead of us unnecessarily deleting and creating the sub directory again.

Reply:
Issue Created here: https://dev.azure.com/fluidframework/internal/_workitems/edit/2205